Geometry/CSCGeometryBuilder/src/CSCGeometryParsFromDD.cc

```diff
@@ -301,3 +460,366 @@ bool CSCGeometryParsFromDD::build(const DDCompactView* cview,
   }
   return true;
 }
+
+// dd4hep
+bool CSCGeometryParsFromDD::build(const cms::DDCompactView* cview,
+                                  const cms::MuonNumbering& muonConstants,
+                                  RecoIdealGeometry& rig,
+                                  CSCRecoDigiParameters& rdp) {
+  const std::string attribute = "MuStructure";
+  const std::string value = "MuonEndcapCSC";
+  cms::DDFilteredView fv(cview->detector(), cview->detector()->worldVolume());
+  cms::DDSpecParRefs refs;
+  const cms::DDSpecParRegistry& mypar = cview->specpars();
+  mypar.filter(refs, attribute, value);
+  fv.mergedSpecifics(refs);
+
+  int noOfAnonParams = 0;
+
+  std::vector<double> uparvals;
+  std::vector<double> fpar;
+  std::vector<double> dpar;
+  std::vector<double> gtran(3);
+  std::vector<double> grmat(9);
+  std::vector<double> trm(9);
+  while (fv.firstChild()) {
+    MuonBaseNumber mbn = muonConstants.geoHistoryToBaseNumber(fv.history());
+
+    cms::CSCNumberingScheme cscnum(muonConstants.values());
+
+    cscnum.baseNumberToUnitNumber(mbn);
+    int id = cscnum.getDetId();
+
+    CSCDetId detid = CSCDetId(id);
+    int jendcap = detid.endcap();
+    int jstation = detid.station();
+    int jring = detid.ring();
+    int jchamber = detid.chamber();
+    int jlayer = detid.layer();
+
+    // Package up the wire group info as it's decoded
+    CSCWireGroupPackage wg;
+    uparvals.clear();
+
+    // if the specs are made no need to get all this crap!
+    int chamberType = CSCChamberSpecs::whatChamberType(jstation, jring);
+
+    size_t ct = 0;
+    bool chSpecsAlreadyExist = false;
+
+    for (; ct < rdp.pChamberType.size(); ++ct) {
```

Comment:
this loop doesn't do anything

Reply:
It is present in the DD side of the code too. There is an "if" command and  a "break" command inside the loop. I don't know if it is right to delete these lines